package/gluon-core/luasrc/lib/gluon/upgrade/115-swconfig

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/lua
+
+local platform = require 'gluon.platform'
+local uci = require('simple-uci').cursor()
+
+local function get_keys(t)
+	local keys={}
+	local n=0
+
+	for k,_ in pairs(t) do
+		n=n+1
+		keys[n]=k
+	end
+	return keys
+end
+
+local switch_vlans = {
+	-- device identifier, lan ports, wan ports
+	["ubnt,unifiac-pro"]={"2 0t", "3 0t"},
+}
+
+local devices = get_keys(switch_vlans)
+
+
+if platform.match('ath79', 'generic', devices) then
+	local board_name = platform.get_board_name()
+
+	local assignments = switch_vlans[board_name]
+
+--	if device_required_vlan_config then
```

Comment:
In case this should be a guard to only apply the workaround on first setup, this should happen at the beginning of the script.

OTOH, i see no downside in executing this script everytime

Reply:
In that case I'd keep it simple and go with the latter.

Reply:
As the network config is reset on each reconfigure, there is no difference between the first and later runs.